student_and_teacher.rb

```diff
@@ -0,0 +1,47 @@
+class Person
+  attr_reader :id
+
+  attr_accessor :age, :name
+
+  def initialize(age, name = 'unknown', parent_permission: true)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
+
+  def can_use_services?
+    of_age? || @parent_permission
+  end
+
+  private
+
+  def of_age?
+    @age >= 18
+  end
+end
+
+class Student < Person
+  attr_accessor :classroom
+
+  def initialize(age, class_room, name = 'unknown', parent_permission: true)
+    super(name, age, parent_permission)
+    @class_room = class_room
+  end
+
+  def play_hooky
+    '¯(ツ)/¯'
+  end
+end
+
+class Teacher < Person
+  attr_accessor :specialization
+
+  def initialize(age, specialization, name = 'unknown', parent_permission: true)
+    super(name, age, parent_permission)
+    @specialization = specialization
+  end
+
+  def can_use_services?
+    true
+  end
+end
```

Comment:
- [ ] Great Job implementing the student and teacher class, but as required by the project requirements, each class should be saved in a separate file. Please kindly make changes

<img width="396" alt="image" src="https://github.com/sam2if/school_library_people/assets/52934128/0afa65cf-269e-4b91-8f8d-23f7838a94fa">